paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java

```diff
@@ -1110,54 +1088,57 @@ CommitResult tryCommitOnce(
                     e);
         }
 
-        if (commitSnapshotImpl(newSnapshot, deltaStatistics)) {
-            LOG.info(
-                    "Successfully commit snapshot {} to table {} by user {} "
-                            + "with identifier {} and kind {}.",
+        boolean success;
+        try {
+            success = commitSnapshotImpl(newSnapshot, deltaStatistics);
+        } catch (Exception e) {
+            // commit exception, not sure about the situation and should not clean up the files
+            LOG.warn("Retry commit for exception.", e);
+            return new RetryResult(latestSnapshot, baseDataFiles, e);
```

Comment:
When committing `APPEND` snapshot, `conflictCheck` will be a `noConflictCheck`. Let's say `commitSnapshotImpl` throws an exception, but the commit actually succeeds. If we retry from `tryCommit`, new files will not be checked for conflicts, and the same new file might be committed twice.